### ISSUE TITLE
Clip images on the proper Area of Interest

### ISSUE THREAD

**Issue by ranghetti** [ranghetti/sen2r#337]
_Message received by email (translated from Italian with Google translate)_

We did several tests to start downloading Sentinel-2 images using the gui, but we encountered a problem. We would like to be able to download Sentinel images already cropped for our ROI and not the whole images (clip_on_extent function), as it is a multi-year study and unfortunately we have limited storage space.
We tried to load the shape file in the "Area of ​​interest" section but we noticed that it still downloads the whole tiles, and not cropped according to the vector we loaded (a series of buffers of about 50 m radius).
Is it through the "Load a vector" function that you also define the area for the clip or do you have to do it differently?
Our polygons are spread over three tiles covering a large part of Piedmont and the Aosta Valley.
Finally, we would like to have a time series of images of three complete years and then access the Long Term Archive (pre-2018), is there the possibility to download all the images starting from January 1, 2017?

**Comment by ranghetti** [comment]
Hi,

original Sentinel-2 archives are always downloaded as whole tiles, since images are packed as tiles in the original file format. 
After downloading them, the `sen2r()` argument `clip_on_extent` allows defining the extent (**bounding box**) of the output products. If you have loaded a vector file containing detached and distant polygons, the resulting bounding box could be large: in this case, to reduce output file size you can alternatively:
1. set also `extent_as_mask = TRUE`: in this case, a single file for each date will be produced, and the area outside polygons will be set to NA; or
2. launch separate sen2r calls using, within each call, a different `extent` (a vector file containing a single polygon) and 'extent_name`: in this case, multiple files will be produced for each date (one per polygon) (see i.e. [this example](https://sen2r.ranghetti.info/articles/sen2r_cmd.html#load-parameters-from-a-json-file-changing-some-of-them-in-the-call-to-sen2r)).

To save disk usage, orignal downloaded SAFE archives should be removed after use (using `rm_safe = "yes"`). Currently, there is an open issue on the {sen2r} dependency {sf} which does not allow doing that on Windows (see #336); so, the archives heve to be removed manually until this will be solved by {sf} maintainers.

Finally, regarding Long Term Archive: {sen2r} allows downloading from it (see [this post](https://luigi.ranghetti.info/post/order-s2-lta/)), but you have to consider the current limitations affecting this infrastructure (see #314).

**Comment by sravetto** [comment]
Hi ranghetti,
I have a further question on this issue.
I would like to download a large amount of images, clipped on a limited extension (identified by a shape).
Since the number of images is large, I would like to download SAFE -> clip it -> delete it (while maintaining the processed files; in my case BOA corrected and SCL) -> go through the second, and so on. Aiming at this, I set safe_dir <- tempfile() and, within the script,   processing_order= "by_date",  rmtmp = TRUE.
I succesfully uploaded the vector file to sen2r and the workflow apparently works.
However, in my output folder I find two folders: BOA and SCL. BOA (i.e. where I expected to find my BOA corrected .tiffs) is empty, while SCL is ok... 
My BOA corrected files are located in C:\Users\root\AppData\Local\Temp\RtmpwJclke\sen2r_1e604a141ec0\group1e603f797733\warped\BOA 
and then, deleted after being clipped!!
I correctly set path_out = "T:\\myfolder"
How can I fix this? 
Many thanks for any help!!
SR
Here you can find my script, if needed:

safe_dir <- tempfile()

out1 <- sen2r(
  gui = FALSE,
  timewindow = c("2017-08-01", "2020-08-27"),
  preprocess = TRUE,
  s2_levels = "l2a",
  sel_sensor = c("s2a", "s2b"),
  online = TRUE,
  order_lta = TRUE,
  downloader = "builtin",
  overwrite_safe = TRUE,
  rm_safe="yes",
  max_cloud_safe = 100,
  step_atmcorr = "l2a",
  extent = buffer,
  extent_name = "buffer",  
  list_prods = c("BOA","SCL"),
  list_indices = NA,
  list_rgb = NA,
  mask_type = "cloud_and_shadow",
  mask_buffer = 20,  
  max_mask = 10,
  clip_on_extent= TRUE,
  extent_as_mask= TRUE,
  path_out = "T:\\myfolder",
  path_l2a = safe_dir,
  processing_order= "by_date",
  rmtmp = TRUE,
  parallel = TRUE
)

**Comment by ranghetti** [comment]
This is the expected behaviour: the *warped* BOA images are a temporary product (an intermediate `sen2r()` step: the geometric operations have been done, but the cloud mask must still be applied), so they are saved in a temporay directory and deleted after cloud masking; instead, the *warped* SCL images are an output product (because SCL images are never cloud masked), so they are saved in the output folder.

You are missing the output BOA images because something is going wrong in the cloud masking step. What about using the command you posted, but with `sen2r(..., mask_buffer = 0, ...)`? If in this case you will obtain the desired BOA images, the reason of the error would be that you did not correctly configure an external GDAL environment (which is required to smooth and bufferise cloud masks). In this case, carefully read this section: https://sen2r.ranghetti.info/articles/installation.html#installing-other-dependencies-generally-unrequired-and-discouraged.

**Comment by sravetto** [comment]
Many thanks, mask_buffer = 0 works!!!

Just another question: as I have to download a lot of images from LTA, I used the script s2_order. Anyway, it outputs "[2020-09-08 13:50:20] 59 of 1159 Sentinel-2 images were correctly ordered. [...] [2020-09-08 13:50:20] 1100 of 1159 Sentinel-2 images were not correctly ordered (HTML status code: 403200) because user 'sravetto01' offline products retrieval quota exceeded.". I created other accounts on Copernicus, but each of them allows me to order just 59 images... 
Is there any solution to this issue? Am I doing something wrong?
Thank you again for your help...
simone
